src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -827,17 +827,18 @@ public INamedTypeSymbol GetTypeByMetadataName(string fullyQualifiedMetadataName)
         protected abstract INamedTypeSymbol CommonGetTypeByMetadataName(string metadataName);
 
         /// <summary>
-        /// Returns a new INamedTypeSymbol with the given element types and (optional) element names.
+        /// Returns a new INamedTypeSymbol with the given underlying type and (optional) element names.
         /// </summary>
         /// <remarks>
         /// Since VB doesn't support tuples yet, this call will fail in a VB compilation.
+        /// Also, the underlying type needs to be tuple-compatible.
         /// </remarks>
-        public INamedTypeSymbol CreateTupleTypeSymbol(ImmutableArray<ITypeSymbol> elementTypes, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
+        public INamedTypeSymbol CreateTupleTypeSymbol(INamedTypeSymbol underlyingType, ImmutableArray<string> elementNames = default(ImmutableArray<string>))
```

Comment:
@CyrusNajmabadi Given the current design, we cannot hide this API, it is needed for correctness. We need to know Identity of the underlying type an other its constituents in order to construct exact tuple type symbol that consumer needs. For cases when consumer prefers us to find the first suitable underlying type, we can provide another overload that doesn't take the underlying type, but take element types instead.